tools/scripts/internal-link-checker.js

```diff
@@ -0,0 +1,24 @@
+const { execSync } = require('child_process');
+const dotenv = require('dotenv');
+
+dotenv.config();
+
+const internalLinks = process.env.INTERNAL_LINKS.split(',');
+
+if (internalLinks.length === 0) {
+  process.exit(1);
+}
+
+for (const internallink of internalLinks) {
+  const result = execSync(
+    `grep --include="./dist/**.json" -rl "./dist" -e "${internallink}"`,
+  );
+  if (result.toString().length > 0) {
+    console.log(
+      `Failed because internal links were found in files:\n${result.toString()}`,
+    );
+    process.exit(1);
```

Comment:
If there are multiple internal link domains configured, and it detects the first one, it will fail already. 
After fixing the first one it might run into another error for the second match in a consecutive CI run. 

I'm not saying we need to improve this, but it could be handled by collecting the results in an array outside of the loop, and having a check after the loop if the result array contains errors.
This way every domain match violation would be caught in a single run and can be fixed as one as well.